sync-online/service-worker.js

```diff
@@ -0,0 +1,145 @@
+// [Working example](/serviceworker-cookbook/sync-online/).
+
+var CACHE_NAME = 'dependencies-cache';
+
+// Files required to make this app work offline
+var REQUIRED_FILES = [
+  '/',
```

Comment:
I think the scope only defines which URLs are controlled by the service worker. The URLs of the requests should be relative to the service worker script URL.

'/' is '/sync-online/' for both.